drivers/metal/metal_device_properties.mm

```diff
@@ -261,8 +277,14 @@
 	}
 
 	limits.maxPerStageBufferCount = 31;
-	limits.maxPerStageSamplerCount = 16;
-	if ([p_device supportsFamily:MTLGPUFamilyApple6]) {
+	if (supportsFamilyMetal3) {
+		limits.maxPerStageSamplerCount = 1024;
+	} else {
+		limits.maxPerStageSamplerCount = 16;
+	}
+	if (supportsFamilyMetal3) {
+		limits.maxPerStageTextureCount = 1024;
```

Comment:
Are you sure? It seems to be `16` and `128` for as well.

https://developer.apple.com/metal/Metal-Feature-Set-Tables.pdf

Reply:
Oops, I read the wrong numbers.